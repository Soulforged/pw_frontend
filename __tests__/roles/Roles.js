//@flow
import React from "react";
import { shallow } from "enzyme";
import Roles from "src/roles/Roles";

it("matches crud structure", () => {
  const props = {
    openForm: jest.fn()
  };
  const wrapper = shallow(<Roles {...props} />);
  expect(wrapper.find("button")).not.toBe(null);
  const button = wrapper.find("button");
  const wrapperProps = button.props();
  wrapperProps.onClick();
  expect(props.openForm).toHaveBeenCalledTimes(1);
});
