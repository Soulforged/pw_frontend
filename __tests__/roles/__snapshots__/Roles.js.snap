// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="pnls"
    id="main-pnl"
  >
    <div>
      <h4
        className="form-header trebuchet bold"
      >
        User Role(s) List
        <button
          className="add-new-btn pointer pull-right bold"
        >
          <i
            className="fa fa-plus theme"
          />
          New Role
        </button>
      </h4>
      <div
        className="tbl-wrapper"
      >
        <lifecycle(branch(Component))
          columns={
            Array [
              Object {
                "Header": "ID",
                "accessor": "id",
                "maxWidth": 50,
              },
              Object {
                "Header": "Role Name",
                "accessor": "name",
              },
              Object {
                "Cell": [Function],
                "Header": "Status",
                "accessor": "status",
              },
            ]
          }
          createButtonTitle="New Role"
          filter={false}
          title="User Role(s) List"
        />
      </div>
    </div>
  </div>
</div>
`;
